pilot/pkg/config/kube/gateway/controller.go

```diff
@@ -65,72 +66,298 @@ var errUnsupportedOp = fmt.Errorf("unsupported operation: the gateway config sto
 type Controller struct {
 	// client for accessing Kubernetes
 	client kube.Client
-	// cache provides access to the underlying gateway-configs
-	cache model.ConfigStoreController
-
-	// Gateway-api types reference namespace labels directly, so we need access to these
-	namespaces       kclient.Client[*corev1.Namespace]
-	namespaceHandler model.EventHandler
-
-	// Gateway-api types reference secrets directly, so we need access to these
-	credentialsController credentials.MulticlusterController
-	secretHandler         model.EventHandler
 
 	// the cluster where the gateway-api controller runs
-	cluster cluster.ID
-	// domain stores the cluster domain, typically cluster.local
-	domain string
-
-	// state is our computed Istio resources. Access is guarded by stateMu. This is updated from Reconcile().
-	state   IstioResources
-	stateMu sync.RWMutex
+	cluster  cluster.ID
+	revision string
 
 	// statusController controls the status working queue. Status will only be written if statusEnabled is true, which
 	// is only the case when we are the leader.
-	statusController *atomic.Pointer[status.Controller]
-
-	tagWatcher revisions.TagWatcher
+	status *StatusCollections
 
 	waitForCRD func(class schema.GroupVersionResource, stop <-chan struct{}) bool
+	outputs    Outputs
+
+	gatewayContext krt.RecomputeProtected[*atomic.Pointer[GatewayContext]]
```

Comment:
Can you remind me what RecomputeProtected is and why we use it here and for tag watcher?